hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/scm/client/OMBlockPrefetchClient.java

```diff
@@ -0,0 +1,369 @@
+
+          // Return unused valid blocks back to the front of the queue (preserving original order).
+          for (int i = tempValidBlocks.size() - 1; i >= 0; i--) {
+            queue.addFirst(tempValidBlocks.get(i));
+          }
+          metrics.incrementCacheMisses();
+        } else {
+          for (ExpiringAllocatedBlock expiringBlock : tempValidBlocks) {
+            AllocatedBlock block = expiringBlock.getBlock();
+            List<DatanodeDetails> sortedNodes =
+                sortDatanodes(block.getPipeline().getNodes(), clientMachine, clusterMap);
+            if (!Objects.equals(sortedNodes, block.getPipeline().getNodesInOrder())) {
+              block = block.toBuilder()
+                  .setPipeline(block.getPipeline().copyWithNodesInOrder(sortedNodes))
+                  .build();
+            }
+            allocatedBlocks.add(block);
+          }
+          metrics.incrementCacheHits();
+        }
+
+        int queueSize = queue.size();
+        if (queueSize < minBlocks) {
+          int blocksToPrefetch = maxBlocks - queueSize;
+          LOG.debug(
+              "Cache for {} is below threshold (size: {}, min: {}). Submitting async prefetch task for {} blocks.",
+              replicationConfig, queueSize, minBlocks, blocksToPrefetch);
+          submitPrefetchTask(scmBlockSize, blocksToPrefetch, replicationConfig, serviceID);
+        }
+
+        metrics.addReadFromQueueLatency(Time.monotonicNowNanos() - readStartTime);
+        return allocatedBlocks;
+
+      } else {
+        LOG.debug("Bypassing cache for {}. Reason: {}", replicationConfig, queue == null ?
+            "Unsupported replication config for caching." : "ExcludeList provided.");
+        metrics.addReadFromQueueLatency(Time.monotonicNowNanos() - readStartTime);
+        metrics.incrementCacheMisses();
+        return scmBlockLocationProtocol.allocateBlock(scmBlockSize, numBlocks, replicationConfig, serviceID,
+            excludeList, clientMachine);
+      }
+    } else {
+      return scmBlockLocationProtocol.allocateBlock(scmBlockSize, numBlocks, replicationConfig, serviceID, excludeList,
+          clientMachine);
+    }
+  }
+
+  private void submitPrefetchTask(long blockSize, int blocksToPrefetch, ReplicationConfig repConfig, String serviceID) {
+
+    if (!isPrefetching.compareAndSet(false, true)) {
+      LOG.debug("Prefetch already in progress. Skipping new task for {}.", repConfig);
+      return;
+    }
+
+    if (prefetchExecutor == null || prefetchExecutor.isShutdown()) {
+      LOG.warn("Async prefetch executor is not running or shutdown. Skipping prefetch task for {}.", repConfig);
+      return;
+    }
+
+    prefetchExecutor.submit(() -> {
+      try {
+        List<AllocatedBlock> prefetchedBlocks = captureLatencyNs(metrics.getPrefetchLatencyNs(),
+            () -> scmBlockLocationProtocol.allocateBlock(blockSize, blocksToPrefetch,
+            repConfig, serviceID, (ExcludeList) Collections.emptyList(), null));
```

Comment:
```suggestion
            repConfig, serviceID, new ExcludeList(), null));
```

Reply:
> instead of new ExlucludeList(), use Collections.emptyList()

Made the above change as per the review comment: [link](https://github.com/apache/ozone/pull/7550#discussion_r2123120278), committed the suggestion.